Library/ViewModels/CommentsViewModel.swift

```diff
@@ -177,12 +179,7 @@ public final class CommentsViewModel: CommentsViewModelType,
       .map(unpack)
       .map(commentsReplacingCommentById)
 
-    self.resetCommentComposerAndScrollToTop = commentsWithFailableOrComment
-      // We only want to scroll to top for failable comments as they represent the initial
-      // comment that's inserted. We check here to see that the first comment's ID is a UUID
-      // to determine this. There may be better ways.
-      .map { UUID(uuidString: $0.first?.id ?? "") }
-      .filter(isNotNil)
+    self.resetCommentComposerAndScrollToTop = self.commentComposerDidSubmitTextProperty.signal.skipNil()
```

Comment:
I've changed this so that we don't actually wait for the failable comment to be created, simply knowing the the user posted text should be enough to know that it _will_ be created and placed at the top. This also reliably always clears the textfield.